src/main/java/com/google/devtools/build/lib/analysis/starlark/StarlarkRuleClassFunctions.java

```diff
@@ -402,6 +403,9 @@ public StarlarkCallable rule(
       }
       builder.addExecGroups(execGroupDict);
     }
+    if (test && (execGroupDict == null || !execGroupDict.containsKey(TEST_RUNNER_EXEC_GROUP))) {
```

Comment:
The actual `execGroups` data in `RuleClass.Builder` is a map: https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/packages/RuleClass.java;drc=8965d25af90ba975cd207c84fa461b499d8d51c9;l=820

Would it be easier to expose a `Set<String> getExecGroupNames()` or `boolean hasExecGroup(String name)` method on `RuleClass.Builder`?

This works but it's not elegant and requires keeping track of the existing groups. It's fine for Starlark, but native rules can have inheritance which would make this trickier.

Reply:
Done. I was wondering about the same thing – my reasoning against this was that I didn't see getters on any other builder classes, and I wasn't sure what you'd think of it.